File: app/src/main/java/mishka/rssreader/di/module/FeedModule.java
package mishka.rssreader.di.module;

import dagger.Module;
import dagger.Provides;
import mishka.rssreader.data.DataManager;
import mishka.rssreader.data.SimpleDataManager;
import mishka.rssreader.ui.feed.FeedAdapter;
import mishka.rssreader.ui.feed.FeedViewModel;

@Module
public class FeedModule {

    @Provides
    FeedAdapter provideFeedAdapter(){
        return new FeedAdapter();
    }

    @Provides
    FeedViewModel provideFeedViewModel(DataManager dataManager){
        return new FeedViewModel(dataManager);
    }

}

File: app/src/main/java/mishka/rssreader/di/module/FeedModule.kt
package mishka.rssreader.di.module;

import dagger.Module;
import dagger.Provides;
import mishka.rssreader.data.DataManager;
import mishka.rssreader.data.SimpleDataManager;
import mishka.rssreader.ui.channelsettings.ChannelSettingsViewModel
import mishka.rssreader.ui.feed.FeedAdapter;
import mishka.rssreader.ui.feed.FeedViewModel;
import mishka.rssreader.ui.feed.OnFeedItemClickListener

@Module
class FeedModule(val itemClickListener: OnFeedItemClickListener) {

    @Provides
    fun provideFeedAdapter(): FeedAdapter {
        return FeedAdapter(itemClickListener)
    }

    @Provides
    fun provideFeedViewModel(dataManager: DataManager): FeedViewModel {
        return FeedViewModel(dataManager)
    }

}
